packages/p2p-v2/src/components/Modals/OrderDetailsComplainModal/OrderDetailsComplainModalRadioGroup/OrderDetailsComplainModalRadioGroup.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { RadioGroup } from '@/components';
+import { useDevice } from '@deriv-com/ui';
+import './OrderDetailsComplainModalRadioGroup.scss';
+
+type TOrderDetailsComplainModalRadioGroupProps = {
+    disputeReason: string;
+    isBuyOrderForUser: boolean;
+    onCheckboxChange: (reason: string) => void;
+};
+
+const OrderDetailsComplainModalRadioGroup = ({
+    disputeReason,
+    isBuyOrderForUser,
+    onCheckboxChange,
+}: TOrderDetailsComplainModalRadioGroupProps) => {
+    const { isMobile } = useDevice();
+    return (
+        <RadioGroup
+            className='p2p-v2-order-details-complain-modal-radio-group'
+            name='reason'
+            onToggle={event => onCheckboxChange(event.target.value)}
+            required
+            selected={disputeReason}
+            textSize={isMobile ? 'md' : 'sm'}
+        >
+            <RadioGroup.Item
+                label={
+                    isBuyOrderForUser
+                        ? 'I’ve made full payment, but the seller hasn’t released the funds.'
+                        : 'I’ve not received any payment.'
+                }
+                value={isBuyOrderForUser ? 'seller_not_released' : 'buyer_not_paid'}
+            />
+            <RadioGroup.Item
+                label={
+                    isBuyOrderForUser
+                        ? 'I wasn’t able to make full payment.'
+                        : 'I’ve received less than the agreed amount.'
+                }
+                value='buyer_underpaid'
+            />
+            <RadioGroup.Item
+                label={
+                    isBuyOrderForUser
+                        ? 'I’ve paid more than the agreed amount.'
+                        : 'I’ve received more than the agreed amount.'
+                }
+                value='buyer_overpaid'
+            />
+            <RadioGroup.Item
+                hidden={isBuyOrderForUser}
+                label='I’ve received payment from 3rd party.'
+                value='buyer_third_party_payment_method'
+            />
+        </RadioGroup>
+    );
+};
```

Comment:
could we use map to display the Radio Items? This is more clean and efficient, easily scalable and maintainable if we were to add any more items in the list 

```suggestion
    const { isMobile } = useDevice();
    const radioItems = [
          {
            hidden: false,
            label: isBuyOrderForUser
              ? 'I’ve made full payment, but the seller hasn’t released the funds.'
              : 'I’ve not received any payment.',
            value: isBuyOrderForUser ? 'seller_not_released' : 'buyer_not_paid',
          },
          {
            hidden: false,
            label: isBuyOrderForUser
              ? 'I wasn’t able to make full payment.'
              : 'I’ve received less than the agreed amount.',
            value: 'buyer_underpaid',
          },
          {
            hidden: false,
            label: isBuyOrderForUser
              ? 'I’ve paid more than the agreed amount.'
              : 'I’ve received more than the agreed amount.',
            value: 'buyer_overpaid',
          },
          {
            hidden: isBuyOrderForUser,
            label: 'I’ve received payment from 3rd party.',
            value: 'buyer_third_party_payment_method',
          },
  ];

  return (
    <RadioGroup>
      {radioItems.map(item => (
        <RadioGroup.Item
          key={item.label}
          label={item.label}
          value={item.value}
          hidden={item.hidden}
        />
      ))}
    </RadioGroup>
  );
};
```